src/java/org/apache/cassandra/index/sai/iterators/KeyRangeLazyIterator.java

```diff
@@ -16,20 +16,22 @@
  * limitations under the License.
  */
 
-package org.apache.cassandra.index.sai.utils;
+package org.apache.cassandra.index.sai.iterators;
 
 import java.io.IOException;
 import java.util.function.Supplier;
 
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+
 /**
  * Delays creating an iterator to the first use.
  */
-public class LazyRangeIterator extends RangeIterator
+public class KeyRangeLazyIterator extends KeyRangeIterator
```

Comment:
Nit: wouldn't it read better as `LazyKeyRangeIterator`?

Reply:
Ah, but ASF always puts `KeyRange*` first, so I guess it makes sense this way.

Reply:
>Ah, but ASF always puts KeyRange* first, so I guess it makes sense this way.

This was my rational to change the name in this way.